ui/src/datatypes.vue

```diff
@@ -103,32 +115,54 @@ export default {
 
         get_datatypes(prefix) {
             if(!this.datatypes) return false;
-            return this.datatypes.filter(d=>{
-                if(~d.name.indexOf(prefix)) {
-                    return true;
-                }
-                return false;
-            }); 
+            if(this.filtered.length){
+                return this.filtered.filter(d=>~d.name.indexOf(prefix));
+            }
+            return this.datatypes.filter(d=>~d.name.indexOf(prefix));
         },
 
         get_not_datatypes(prefix) {
             if(!this.datatypes) return false;
-            return this.datatypes.filter(d=>{
-                if(!~d.name.indexOf(prefix)) {
-                    return true;
-                }
-                return false;
-            }); 
+            if(this.filtered.length){
+                return this.filtered.filter(d=>!d.name.includes(prefix));
+            }
+            return this.datatypes.filter(d=>!d.name.includes(prefix));
         },
 
+        clearQuery() {
+            this.query = ''
+            this.changeQuery();
+        },
+
+        changeQueryDebounce() {
+            clearTimeout(queryDebounce);
+            queryDebounce = setTimeout(this.changeQuery, 300);        
+        },
+
+        changeQuery() {
+            if(!this.datatypes) return setTimeout(this.changeQuery, 300);
+            //document.location="#"; //clear hash //TODO what is this?
+            sessionStorage.setItem("datatypes.query", this.query);
+            this.load();
+        },
+
+        load(){
+            if(this.query){
+                this.query.split(" ").forEach(q=>{
+                    if(q == "") return;
+                    this.filtered = this.datatypes.filter((datatype) =>
```

Comment:
will it be better to use `datatype => datatype.name.toLowerCase().includes(q.toLowerCase()))` in one single line or is this okay ?

Reply:
I am using a filtered array to store the filtered datatypes from all the data types we have and when the query is empty it should show the original datatypes.

Reply:
Yes, but I think it will only take the last token into an account.. right? Is that how you want it to work?

Reply:
I want it to filter results with every key press and backspace pressed and if the query becomes empty or null then to make it return the original/unfiltered datatypes

Reply:
if you enter "mp2rage life eeg" in the query, you will see *all* datatypes that names matches the eeg - but I would think user would expect datatype that has all 3 tokens matching. Are you aware that `this.filtered` is overwritten for each forEach loop - effectively making the search to be performed on the last token?

Reply:
Ohh now I see! I'll fix that asap